nexus/src/authz/api_resources.rs

```diff
@@ -38,15 +112,19 @@ use uuid::Uuid;
 /// You can perform authorization checks on children of the Fleet (e.g.,
 /// Organizations) using the methods below that return authz objects
 /// representing those children.
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+#[derive(Clone, Copy, Debug)]
 pub struct Fleet;
 /// Singleton representing the [`Fleet`] itself for authz purposes
 pub const FLEET: Fleet = Fleet;
 
 impl Fleet {
     /// Returns an authz resource representing a child Organization
-    pub fn organization(&self, organization_id: Uuid) -> Organization {
-        Organization { organization_id }
+    pub fn organization(
+        &self,
+        organization_id: Uuid,
+        lookup_type: LookupType,
```

Comment:
In order to produce the type-specific 404 errors (`Error::ObjectNotFound`), the `authz::` resource structs now have a `LookupType` that describes how they were looked up in the first place.

For several types, this meant we could no longer derive `Eq` and `PartialEq`.  Those traits are only used for types for which we do an equality check in the Polar file:
https://github.com/oxidecomputer/omicron/blob/2e5c819883c01636f4030e23ebf081891c4272d3/nexus/src/authz/omicron.polar#L204-L212

For some types, I removed these impls because we didn't need them.  For others, I impl'd them by hand in a way that ignores the new LookupType field.